jaxlib/pytree.cc

```diff
@@ -111,6 +111,9 @@ class PyTreeDef {
   // Flattens a Pytree into a list of leaves and a PyTreeDef.
   static std::pair<py::list, std::unique_ptr<PyTreeDef>> Flatten(py::handle x);
 
+  // Tests whether the given list is a flat list of leaves.
+  static bool AllLeaves(const py::iterable& x);
```

Comment:
nit: maybe the name `AllLeaves` could be confused with a function that returns all leaves. Or perhaps not, due to the type? I am wondering if `AreAllLeaves` is clearer?